### PR TITLE
File system

### PR DESCRIPTION
**Pull request by ahmedqarmout2**
Implementing a tree structure in the backend to save files related to questions and users

### CODE REVIEW

File: server/common.js [hunk 1]
```diff
@@ -137,11 +157,42 @@ var isEmptyObject = function(obj) {
...
 function getDate() {
     return date().format('YYYY-MM-DD hh:mm:ss A');
 }
 exports.getDate = getDate;
+// </Global Function> -----------------------------------------------
+
+// <File System functions> ------------------------------------------
+// make a directory given the path and the name of the new directory
+var mkdir = function (parentPath, directoryName, callback) {
+    var fullPath = path.join(parentPath, directoryName);
```
Comment (mehtadarshan1) [inline server/common.js:204]: are these variables given by the user? 
Don't want any shell hijacking happening here. If it is, please validate them before executing any code.

**Review by esteeroramy — APPROVED** [review]

File: server/common.js (hunk 1 above)
Comment (ahmedqarmout2) [reply server/common.js:204]: good point. how can we check this? This function is backend only, but the other function do use some user input

**Review by Mohammed-JS-Khan — APPROVED** [review]
I like this. Nice work.

**Review by Mohamed-A-Mohamed — APPROVED** [review]
Looks good to me :+1:

File: server/common.js (hunk 1 above)
Comment (mehtadarshan1) [reply server/common.js:204]: Check by seeing if the code you will execute has any special characters in it. It should only be alphabet strings. Disregard any Pipes, slashes, periods, etc...

**Pull request merged**